lib/Crypto/Cipher/AES.pyi

```diff
@@ -12,36 +13,217 @@ from Crypto.Cipher._mode_gcm import GcmMode
 from Crypto.Cipher._mode_siv import SivMode
 from Crypto.Cipher._mode_ocb import OcbMode
 
-AESMode = int
-
-MODE_ECB: AESMode
-MODE_CBC: AESMode
-MODE_CFB: AESMode
-MODE_OFB: AESMode
-MODE_CTR: AESMode
-MODE_OPENPGP: AESMode
-MODE_CCM: AESMode
-MODE_EAX: AESMode
-MODE_GCM: AESMode
-MODE_SIV: AESMode
-MODE_OCB: AESMode
-
-Buffer = Union[bytes, bytearray, memoryview]
-
-def new(key: Buffer,
-        mode: AESMode,
-        iv : Buffer = ...,
-        IV : Buffer = ...,
-        nonce : Buffer = ...,
-        segment_size : int = ...,
-        mac_len : int = ...,
-        assoc_len : int = ...,
-        initial_value : Union[int, Buffer] = ...,
-        counter : Dict = ...,
-        use_aesni : bool = ...) -> \
-        Union[EcbMode, CbcMode, CfbMode, OfbMode, CtrMode,
-              OpenPgpMode, CcmMode, EaxMode, GcmMode,
-              SivMode, OcbMode]: ...
+if sys.version_info >= (3, 8):
+    MODE_ECB: Literal[1]
+    MODE_CBC: Literal[2]
+    MODE_CFB: Literal[3]
+    MODE_OFB: Literal[5]
+    MODE_CTR: Literal[6]
+    MODE_OPENPGP: Literal[7]
+    MODE_CCM: Literal[8]
+    MODE_EAX: Literal[9]
+    MODE_SIV: Literal[10]
+    MODE_GCM: Literal[11]
+    MODE_OCB: Literal[12]
 
+    AESMode = Union[
+        MODE_ECB,
```

Comment:
Is `AESMode` needed?

Reply:
No, it's not needed. But in the absence of an enum, it serves as a kind of grouping that I think helps make the code somewhat self-documenting.